osu.Game/Screens/Edit/GameplayTest/EditorPlayer.cs

```diff
@@ -133,6 +137,76 @@ protected override void PrepareReplay()
 
         protected override bool CheckModsAllowFailure() => false; // never fail.
 
+        public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
+        {
+            if (e.Repeat)
+                return false;
+
+            switch (e.Action)
+            {
+                case GlobalAction.EditorTestPlayToggleAutoplay:
+                    toggleAutoplay();
+                    return true;
+
+                case GlobalAction.EditorTestPlayToggleQuickPause:
+                    toggleQuickPause();
+                    return true;
+
+                case GlobalAction.EditorTestPlayQuickExitToInitialTime:
+                    quickExit(false);
+                    return true;
+
+                case GlobalAction.EditorTestPlayQuickExitToCurrentTime:
+                    quickExit(true);
+                    return true;
+
+                default:
+                    return false;
+            }
+        }
+
+        public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
+        {
+        }
+
+        private void toggleAutoplay()
+        {
+            if (DrawableRuleset.ReplayScore == null)
+            {
+                var autoplay = Ruleset.Value.CreateInstance().GetAutoplayMod();
+                if (autoplay == null)
+                    return;
+
+                var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, [autoplay]);
```

Comment:
I don't suppose we allow collection expressions / `[]` syntax for defining arrays?

```suggestion
                var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, new[] { autoplay });
```

Reply:
/shrug i wanna be able to use them (and i believe i already have in other places). they're one of the few okay recent additions to the language.

Reply:
I'm not necessarily against it myself (to be fair I was confused for a few minutes at first but then I remembered my Python courses). I think it's fine.